gtfu/src/web/graas-report-web.js

```diff
@@ -0,0 +1,343 @@
+
+    reportImageUrl = `${bucketURL}/graas-report-archive/${selectedAgency}/${selectedAgency}-${selectedDate}.png`;
+    img.src = reportImageUrl;
+
+    p = document.getElementById('download');
+    p.href = reportImageUrl;
+    p.download = `${selectedAgency}-${selectedDate}.png`;
+    p.style.display = "inline-block";
+
+    img.onload = function() {
+        imageWidth = this.naturalWidth
+        imageHeight = this.naturalHeight;
+        scaleRatio = window.innerWidth / imageWidth;
+        console.log("scaleRatio: " + scaleRatio);
+
+        timelineCanvasHeight = timelineHeight * scaleRatio;
+        headerHeight = timelineCanvasHeight + dropdownHeight
+        timelineCtx.canvas.height = timelineCanvasHeight;
+
+        mapHeight = imageHeight - timelineHeight;
+        mapCanvasHeight = mapHeight * scaleRatio;
+        mapCtx.canvas.height = mapCanvasHeight;
+
+        // It would be nice to perform this multiplication while initially loading JSON,
+        // but that occurs before image is loaded and scaleRatio is determined by image size
+        trips.forEach(function (item) {
+            item.map.x *= scaleRatio;
+            item.map.y *= scaleRatio;
+            item.map.width *= scaleRatio;
+            item.map.height *= scaleRatio;
+            item.timeline.x *= scaleRatio;
+            item.timeline.y *= scaleRatio;
+            item.timeline.width *= scaleRatio;
+            item.timeline.height *= scaleRatio;
+        });
+        drawReport();
+    };
+
+    window.addEventListener('resize', function(event) {
+
+        load();
+    });
+
+    document.body.addEventListener('click', function(event) {
+
+        drawReport();
+
+        scrollTop = document.documentElement.scrollTop; // Top of current screen
+        timelineScrollTop = scrollTop + dropdownHeight; // Top of current timeline
+        mapScrollTop = scrollTop + headerHeight;        // Top of current map
+        scrollBottom = scrollTop + windowHeight;        // Bottom of current map
+
+        for (var i = 0; i < trips.length; i++){
+            var searchY = event.pageY;
+            var searchObject;
+
+            // If the click occurs below the header, adjust y value and search map objects
+            if(searchY >= mapScrollTop){
+                searchY -=headerHeight;
+                searchObject = trips[i].map;
+
+            // If the click occurs within the header, adjust y value and search timeline objects
+            } else {
+                searchY -=timelineScrollTop;
+                searchObject = trips[i].timeline;
+            }
+            if (objectContainsPoint(searchObject, event.pageX, searchY)){
+                drawMetadata(trips[i]);
+                selectTrip(mapCtx, trips[i].map);
+                selectTrip(timelineCtx, trips[i].timeline);
+                scrollToTrip(trips[i])
+                break;
+            }
+        }
+    });
+}
+
+function objectContainsPoint(object, x, y){
+    if(x >= object.x){
```

Comment:
`return x >= object.x && x <= object.x + object.width && y >= object.y && y <= object.y + object.height;` is more succinct, functionally equivalent and won't cause branch prediction problems :)

Reply:
Also, it looks like you're not returning anything if any of the conditions fail. Maybe not returning anything in js is somehow the same as returning `false`, but this is poor style.